src/shared/cloudformation/cloudformation.ts

```diff
@@ -284,4 +304,209 @@ export namespace CloudFormation {
+     * Also returns undefined if the property is undefined or a number.
+     * @param property Property value to check
+     * @param template Template object to parse through
+     */
+    export function getStringForProperty(
+        property: string | number | object | undefined,
+        template: Template
+    ): string | undefined {
+        if (validatePropertyType(property, 'string', template)) {
+            if (typeof property === 'string') {
+                return property
+            } else if (isRef(property)) {
+                try {
+                    const forcedProperty = property as Ref
+                    return getReffedString(forcedProperty, template)
+                } catch (err) {
+                    getLogger().debug(err)
+                }
+            }
+        }
+
+        return undefined
+    }
+
+    /**
+     * Gets the numeric value for a property in a template.
+     * If the value is a Ref to a parameter, returns the default value of the ref; this may be undefined.
+     * Also returns undefined if the property is undefined or a string.
+     * @param property Property value to check
+     * @param template Template object to parse through
+     */
+    export function getNumberForProperty(
+        property: string | number | object | undefined,
+        template: Template
+    ): number | undefined {
+        if (validatePropertyType(property, 'number', template)) {
+            if (typeof property === 'number') {
+                return property
+            } else if (isRef(property)) {
+                try {
+                    const forcedProperty = property as Ref
+                    return getReffedNumber(forcedProperty, template)
+                } catch (err) {
+                    getLogger().debug(err)
+                }
+            }
+        }
+
+        return undefined
+    }
+
+    /**
+     * Returns whether or not a property or its underlying ref matches the specified type
+     * Does not validate a default value for a template parameter; just checks the value's type
+     * @param property Property to validate the type of
+     * @param type Type to validate the property's type against
+     * @param template Template object to parse through
+     */
+    function validatePropertyType(
+        property: string | number | object | undefined,
+        type: 'string' | 'number',
+        template: Template
+    ): boolean {
+        if (typeof property === type) {
+            return true
+        } else if (isRef(property)) {
+            // property has a Ref, force it to abide by that shape
+            const forcedProperty = property as Ref
+            const param = getReffedParam(forcedProperty, template)
+            const paramType = param.Type === 'Number' ? 'number' : 'string'
+
+            return paramType === type
+        }
+
+        return false
+    }
+
+    /**
+     * Helper function to get a number from a Ref.
```

Comment:
all functions are helpful, so they never need to be qualified as helpers.
```suggestion
     * Gets a number from a Ref.
```